KAS-2748 extra changes based on comments in KAS-2748-PR-1001

mdv-professional opened the pull request:
Extra changes to the publication proof tabs, based on comments in
[PR-1001](https://github.com/kanselarij-vlaanderen/kaleidos-frontend/pull/1001)
[Erika's comment](https://github.com/kanselarij-vlaanderen/kaleidos-frontend/pull/1001#issuecomment-878362607)

**notes**
- i.v.m. (3.) checkboxes: dit is niet gebeurd. Het "Afgewerkt" vinkje wordt in een andere controller geset dan de selectie. Dit maakt het te complex om de selectie te resetten.
- de time-component van de receivedDate van een document wordt, als deze door de gebruiker is ingevoerd, weergegeven met 00:00 (want date-only). Dit ziet er wat merkwaardig uit.
- Ik heb validatie toegevoegd voor de receivedDate in alle upload-modals. Misschien was het niet nodig als ik de datum elke keer terugzette wanneer deze werd gewist door de gebruiker. Dit heb ik nu gelaten.

Commit:
KAS-2748 extra changes based on comments in KAS-2748-PR-1001 | for proof [pub proof]

Commit:
KAS-2748/PR-1001 extra changes: validation + margin | for proof [pub proof]

Commit:
KAS-2748/PR-1001 (busy) extra changes: show validation errors | for proof [pub proof]

Commit:
KAS-2748/PR-1001 extra changes: show validation errors | for proof [pub proof]

Commit:
KAS-2749/PR-1001 onaandachtzaamheid               | extra [pub proof]

Commit:
KAS-2478/PR-1001 validation class                 | extra [pub proof]

Commit:
KAS-2478/PR-1001 isPresent                        | extra [pub proof]

Commit:
KAS-2748/PR-1001 unlog + fix validation for message                          | extra [pub proof]

ValenberghsSven committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/kaleidos-frontend into feature/KAS-2748-pub-proof--extra

ValenberghsSven commented on app/components/publications/publication/translations/translation-upload-modal.hbs:
```diff
@@ -17,18 +17,20 @@
             <div class="auk-o-grid-col-11">
               <div class="auk-form-group-layout auk-form-group-layout--horizontal">
-                <div class="auk-form-group auk-o-flex--align-start">
+                <div class="auk-form-group auk-o-flex--align-start
+                  {{if this.validators.name.showError "auk-form-group--error"}}">
```
`{{this.validators.name.errorClass}}` ?

ValenberghsSven commented on app/pods/publications/publication/proofs/documents/template.hbs:
```diff
@@ -159,6 +159,7 @@
 {{#if this.isPieceUploadModalOpen}}
   <Publications::Publication::Proofs::ProofUploadModal
     @publicationFlow={{this.publicationFlow}}
+    @source="correction"
```
Alle documenten die je upload in de drukproeven / documenten view is een correctie?
Ik ken de flow van drukproeven niet goed, maar wil je hier geen andere documenten uploaden (vb een bron document voor het aanvragen van een drukproef?

Ook geven we de source hier hardcoded mee op 2 plaatsen.
Maar met de waarde "correction" wordt verder niks gedaan. enkel "proof" heeft een gevolg
Willen we dan werken met een boolean ipv tekst? of met Constants als je toch tekst wilt

erikap replied:
Op je eerste comment: dit klopt. Ze kunnen enkel correcties opladen op de drukproeven/documenten-tab. De 'brondocumenten' worden opgeladen bij de vertalingen en worden daar gemarkeerd als 'voor drukproef'. Die verschijnen dan automatisch op de drukproeven/documenten-tab. Ik zal het label van de button aanpassen naar 'Verbeterde drukproef opladen' om verwarring te vermijden.

erikap replied:
Je hebt wel gelijk dat we met `@source="correction"` en `@source="proof"` twee concepten aan het mixen zijn.

Ik denk dat de flags die we definieren op de upload modal component beter zijn:
- `isReceivedDocument` (`true` of `false`): indien `true` 
  - tonen we het invulveld voor ontvangstdatum
- `isCorrectionDocument` (`true` of `false`): indien `true` 
  - tonen we als titel van de modal 'Verbeterd drukproef uploaden voor publicatie...'
  - een read-only veldje na het uploaden van de file 'Verbeterd' met het groene vinkje zoals in de tabel (dus zoals de verbeterd-toggle in het prototype, maar dan met een read-only vinkje ipv de toggle). Op die manier is het waarschijnlijk duidelijker voor de gebruiker dat ze een verbeterd document aan het opladen zijn. Anders zien ze het verbeterd-vinkje pas verschijnen na het opladen in de tabel.

MikiDi replied:
Opgelost in https://github.com/kanselarij-vlaanderen/kaleidos-frontend/pull/1006/commits/17cc0fb19e3a9e39267758a178470f13c11fa706 en https://github.com/kanselarij-vlaanderen/kaleidos-frontend/pull/1006/commits/79d3168dccd939cd48f89ad7aa6f4112b248f5ac

ValenberghsSven commented on app/components/publications/publication/proofs/proof-upload-modal.hbs:
```diff
@@ -29,6 +29,14 @@
                   <Auk::Label>{{t "file-type"}}</Auk::Label>
                   <Utils::Documents::FileTypePill @extension={{this.file.extension}} />
                 </div>
+                {{#if (eq @source "proof")}}
+                  <div class="auk-form-group {{this.validators.receivedAtDate.errorClass}}">
+                    <Auk::Label for="proofUploadReceivedAtDate">{{t "received-at"}}</Auk::Label>
+                    <Auk::Datepicker id="proofUploadReceivedAtDate"
+                      @date={{this.receivedAtDate}}
+                      @onChange={{this.setReceivedAtDate}} />
+                  </div>
```
een ontvangen drukproef krijgt een received date, maar een gecorrigeerde drukproef niet?
Ben niet helemaal mee met de flow hiervan, kan dus correct zijn.

erikap replied:
Dit klopt. Een ontvangen drukproef krijgt OVRB van een externe partij. Ze willen dan de datum waarop ze dat document ontvangen hebben registeren (want dat is niet noodzakelijk de datum waarop ze het document uploaden).
Het gecorrigeerde drukproef document is een document dat door OVRB zelf opgesteld wordt om opnieuw naar het BS te sturen. Aangezien ze dat document zelf gemaakt hebben en uploaden moeten ze hier geen ontvangsdatum voor registreren.

erikap committed:
Remove unnecessary async on action

erikap committed:
Remove passing of default option on auk-navbar component

erikap committed:
Reuse publication-subcase already fetched in model hook in setupController

erikap committed:
Remove modal state management from setupController

In line with other parts of the application modal state is managed
in the controller only

erikap committed:
Add clarifying comments on documents loading for publication proofs

erikap committed:
Change button label to upload corrected proof document

erikap committed:
Remove non-function actions from proof documents

Functionality to edit and remove proof documents may not be implemented
before the go live. Therefore removing placeholder buttons from template

erikap committed:
Update button label to upload proof document on request activity

erikap commented:
Wijzigingen die ik nog gemaakt hebt zijn niet structureel. Enkel wat button labels aangepast, clarification comments toegevoegd en niet-werkende buttons verwijderd.

Voor de disabled checkboxes: ok om dit nu zo te laten. We zullen wel zien als er feedback komt van de gebruikers dat we dan een nieuwe story maken om dit op te lossen.

Voor de received date: ok om enkel de datum te tonen in de overzichtstabel aangezien de gebruiker enkel de datum kan invullen. Kan je dit op een gelijkaardige manier wijzigen bij vertalingen dan? Want daar lijkt het tijdstip op de huidige tijd gezet te worden.

Wat ik nog zou doen in deze PR:
- de feedback rond `@source="proof"` verwerken (https://github.com/kanselarij-vlaanderen/kaleidos-frontend/pull/1006#discussion_r669578873)
- de ontbrekende error class toevoegen (https://github.com/kanselarij-vlaanderen/kaleidos-frontend/pull/1006#discussion_r669569745)

Commit:
KAS-2748 improvements | extra [pub proof]

Commit:
KAS-2748 improvements | extra [pub proof]

Commit:
Merge branch 'feature/KAS-2748-pub-proof--extra' of https://github.com/kanselarij-vlaanderen/kaleidos-frontend into feature/KAS-2748-pub-proof--extra

Commit:
KAS-2748 merge issue                              | extra [pub proof]

ghost commented:
Dit is nu gebeurd.

**opmerkingen**
- de checkbox heeft nog geen aparte disabled styling. @brenner-company

brenner-company commented:
@mdv-professional Die styling komt er nog, voor mijn verlof start morgenmiddag, aan 👍

brenner-company committed:
Update au-kaledos-css package

brenner-company committed:
Add support for disabled state with checkbox component

brenner-company committed:
Adjust checkbox styleguide docs

brenner-company commented:
@mdv-professional Done. Disabled state kan nu aangepast worden via `@disabled`.

MikiDi committed:
Merge branch 'development' into feature/KAS-2748-pub-proof--extra

MikiDi committed:
use errorclass var instead of hardcoded css class

Commit:
KAS-2748 @disabled                                | extra [pub proof]

ghost commented:
Ook done.

MikiDi committed:
Merge branch 'feature/KAS-2748-pub-proof--extra' of github.com:kanselarij-vlaanderen/kaleidos-frontend into feature/KAS-2748-pub-proof--extra

MikiDi committed:
add doc to auk/checkbox regarding disabled

MikiDi committed:
remove unnecessary initializations

MikiDi committed:
clarify need for elaborate date selection handling

MikiDi committed:
use "this"-scope in templates

MikiDi committed:
use double quotes in templates

MikiDi committed:
add page size where missing because of custom case

MikiDi committed:
double quotes in templates

MikiDi committed:
add success-toast on non-confirmed interaction

MikiDi committed:
Merge branch 'development' into feature/KAS-2748-pub-proof--extra
